src/data_internal.c

```diff
@@ -41,7 +41,8 @@ merge_accumulator_copy_message(merge_accumulator *ma, message msg)
 _Bool
 merge_accumulator_resize(merge_accumulator *ma, uint64 newsize)
 {
-   platform_status rc = writable_buffer_resize(&ma->data, newsize);
+   platform_status rc =
+      writable_buffer_resize(&ma->data, merge_accumulator_length(ma), newsize);
```

Comment:
I don't recall why / what this change is for.  Don't even recall if I added it.
Seems like an inherit from `/main` ?

@rtjohnso -- we should re-visit during review and finalize.